### PR TITLE
DBZ-1419 Support for Connet temporal types in SQL Server

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1419

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java [hunk 1]
```diff
@@ -192,8 +193,21 @@ public static DecimalHandlingMode parse(String value, String defaultValue) {
...
+                                                             .withImportance(Importance.MEDIUM)
+                                                             .withValidation(RelationalDatabaseConnectorConfig::validateTimePrecisionMode)
+                                                             .withDescription("Time, date, and timestamps can be represented with different kinds of precisions, including:"
+                                                                     + "'adaptive_time_microseconds' (the default) like 'adaptive' mode, but TIME fields always use microseconds precision;"
+                                                                     + "'connect' always represents time, date, and timestamp values using Kafka Connect's built-in representations for Time, Date, and Timestamp, "
+                                                                     + "which uses millisecond precision regardless of the database columns' precision.");
+
     private final RelationalTableFilters tableFilters;
 
+    public final TemporalPrecisionMode temporalPrecisionMode;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: Can be `private`?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ public void shouldReceiveChangesForInsertsWithDifferentDataTypes() throws Except
 
         //date and time
         consumer.expects(1);
-        assertInsert(INSERT_DATE_TIME_TYPES_STMT, 1, schemaAndValuesForDateTimeTypes());
+        assertInsert(INSERT_DATE_TIME_TYPES_STMT, 1, schemaAndValuesForDateTimeTypesAdaptiveTimeMicroseconds());
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java]: Why has this changed? Does it indicate a change to the default behavior?

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java (hunk 1 above)
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: @gunnarmorling Fixed, thanks!

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java (hunk 2 above)
Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java]: IIRC we've decided to drop adaptive mode. This was removed in MySQL but was kept in Postgres so this is unification for all three connectors and this has impact on postgres default

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java]: Hum, I can't remember that we decided this. Are you sure?

Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java]: @gunnarmorling https://github.com/debezium/debezium/pull/864/files#diff-d565df219003c24920bee4de756b0324R1172

**Comment by gunnarmorling** [comment]
Ah, well, for MySQL this was actually broken (prone to value overflow); I
don't think it's the same for PostgreSQL, or is it?

**Comment by jpechane** [comment]
Superseded with https://github.com/debezium/debezium/pull/999

**Pull request closed without merging**